lefthook.yml

```
@@ -0,0 +1,11 @@
+pre-push:
+  parallel: true
+
+pre-commit:
+  commands:
+    mypy-check:
+      glob: "*.{py}"
+      run: mypy {staged_files}
+    ruff-check:
+      glob: "*.{py}"
+      run: ruff check {staged_files}
```

also slight preference for when ruff gets fixed automatically when possible, but I leave it up to you
```suggestion
      run: ruff check {staged_files} --fix
```